connect/runtime/src/main/java/org/apache/kafka/connect/runtime/rest/resources/ConnectorsResource.java

```diff
@@ -289,6 +301,38 @@ private void checkAndPutConnectorConfigName(String connectorName, Map<String, St
         return completeOrForwardRequest(cb, path, method, body, null, new IdentityTranslator<T>(), forward);
     }
 
+    private <T> T loadConnectorClass(final String className, final Class<T> type) {
+        try {
+            return type.cast(Class.forName(className).newInstance());
+        } catch (InstantiationException | IllegalAccessException | ClassNotFoundException e) {
+            throw new IllegalStateException(e);
+        }
+    }
+
+    // go through config parameters and replace password field value with "*"
+    // this return a new map, instead of making change to the original object
+    private Map<String, String> maskCredentials(Map<String, String> config) {
```

Comment:
Also, although this method is not public, it's probably useful to put this comment into a JavaDoc comment and to include information about the parameter. Can the `config` parameter be null? Will this method ever return a null value?

Reply:
It looks like after moving this method to the `AbstractHerder`, it's still not available to use in this class. Because the herder in this class is defined as a `Herder` not `AbstractHerder`. So it doesn't know the new method is available.

Should I add another method in the `Herder` interface `connectorMaskedConfig` which returns the masked config and I can implement that in the `AbstractHerder`? Any suggestions?

Reply:
How about defining a new method on `Herder` that safely returns the `ConfigDef` given the name of a connector. This could then be implemented in `AbstractHerder` in terms of the existing methods that properly use the plugin classloaders, and could be called by this class to mask the configuration properties for both a connector and the tasks associated with a connector:

```
private Map<String, String> maskCredentials(String connectorName, Map<String, String> config) {
    ConfigDef configDef = herder.connectorConfigDef(connectorName);
    Map<String, ConfigDef.ConfigKey> definedConfigKeys = configDef.configKeys();
    ...
}
```

I'm not sure there's a way to get around adding a method to `Herder`. It is an interface, but it's not part of the public API and thus part of the implementation.

Reply:
That should work. I'll try that.

Is there a way in the AbstractHerder to get kafka connect config parameters? For backwards compatible we might need to add a parameter "enable.password.mask" for user to opt in. So I want to detect if this config is set to True in the AbstractHerder to decide if we should return the masked password or plain text.

Reply:
@rhauch I'm having a hard time exposing the maskCredentials method from `AbstractHerder` to be used by `ConnectorResource`...

Any suggestions?